main.tf

```diff
@@ -38,47 +50,96 @@ resource "aws_efs_access_point" "default" {
 
   file_system_id = join("", aws_efs_file_system.default.*.id)
 
-  posix_user {
-    gid = var.access_points[each.key]["posix_user"]["gid"]
-    uid = var.access_points[each.key]["posix_user"]["uid"]
-    # Just returning null in the lookup function gives type errors and is not omitting the parameter, this work around ensures null is returned.
-    secondary_gids = lookup(lookup(var.access_points[each.key], "posix_user", {}), "secondary_gids", null) == null ? null : null
+  dynamic "posix_user" {
+    for_each = local.secondary_gids[each.key] != null ? ["true"] : []
+
+    content {
+      gid            = var.access_points[each.key]["posix_user"]["gid"]
+      uid            = var.access_points[each.key]["posix_user"]["uid"]
+      secondary_gids = local.secondary_gids[each.key] != null ? split(",", local.secondary_gids[each.key]) : null
+    }
   }
 
   root_directory {
     path = "/${each.key}"
-    creation_info {
-      owner_gid   = var.access_points[each.key]["creation_info"]["gid"]
-      owner_uid   = var.access_points[each.key]["creation_info"]["uid"]
-      permissions = var.access_points[each.key]["creation_info"]["permissions"]
+
+    dynamic "creation_info" {
+      for_each = try(lookup(var.access_points[each.key]["creation_info"]["gid"], ""), "") != "" ? ["true"] : []
+
+      content {
+        owner_gid   = var.access_points[each.key]["creation_info"]["gid"]
+        owner_uid   = var.access_points[each.key]["creation_info"]["uid"]
+        permissions = var.access_points[each.key]["creation_info"]["permissions"]
+      }
     }
   }
 
   tags = module.this.tags
 }
 
+module "security_group_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+
+  # For backwards compatibility with v0.30.1
+  attributes = concat(module.this.attributes, [var.security_group_suffix])
+
+  context = module.this.context
+}
+
```

Comment:
Users needing backward compatibility should pass in a security group name in its entirety via `security_group_name`. Also
```hcl
  attributes = concat(module.this.attributes, [var.security_group_suffix])
```
is bad style. It should be simply
```hcl
  attributes = [var.security_group_suffix]
```
```suggestion
```

Reply:
security_group_suffix was removed in favor of security_group_name